lib/wovnrb/services/html_converter.rb

```diff
@@ -0,0 +1,226 @@
+    end
+
+    def strip_hreflang(node)
+      supported_langs = @store.settings['supported_langs']
+      if node['hreflang'] && supported_langs.include?(Lang::iso_639_1_normalization(node['hreflang']))
+        node.remove
+      end
+    end
+
+    def add_hreflang
+      !!(@store && @headers)
+    end
+
+    def inject_lang_html_tag
+      lang = @headers.lang_code
+      unless lang == @store.settings['default_lang']
+        @html = @html.sub(/<html\s?([^>]*)?>/i) do |_|
+          if $1.present?
+            "<html lang=\"#{lang}\"" + ' \1' + '>'
+          else
+            "<html lang=\"#{lang}\">"
+          end
+        end
+      end
+    end
+
+    def insert_hreflangs
+      langs = @store.settings['supported_langs'] || []
+
+      # Strip all existing hreflang tags
+      strip_tags_by_regex(hreflang_regex(langs))
+
+      # Insert hreflang tag for each supported language
+      insert_after_tag(parent_tags, hreflang_tags(langs).join)
+    end
+
+    def hreflang_tags(langs)
+      langs.map do |lang|
+        "<link rel=\"alternate\" hreflang=\"#{Lang::iso_639_1_normalization(lang)}\" href=\"#{hreflang(lang)}\">"
+      end
+    end
+
+    def hreflang_regex(langs)
+      /<link [^>]*hreflang=[\"']?#{Regexp.quote(langs.join('|'))}[\"']?(\s[^>]*)?\>/i
+    end
+
+    def parent_tags
+      [
+        /(<head\s?.*?>)/i,
+        /(<body\s?.*?>)/i,
+        /(<html\s?.*?>)/i
+      ]
+    end
+
+    def hreflang(lang_code)
+      # TODO: Refactor to put redirect_location logic to Url class
+      @headers.redirect_location(lang_code)
+    end
+
+    def strip_tags_by_regex(rx)
+      @html = @html.gsub(rx, '')
+    end
+
+    def insert_snippet
+      strip_tags_by_regex(snippet_regex)
+      insert_after_tag(parent_tags, snippet_code)
+    end
+
+    def snippet_regex
+      /<script[^>]*src=[^>]*j\.[^ '\">]*wovn\.io[^>]*><\/script>/i
+    end
+
+    def snippet_code(adds_backend_error_mark = false)
+      snippet_url = 
+        if @store.settings['wovn_dev_mode'].present?
+          '//j.dev-wovn.io:3000/1'
+        else
+          '//j.wovn.io/1'
+        end
+      if adds_backend_error_mark
+        "<script src=\"#{snippet_url}\" data-wovnio=\"#{data_wovnio}\" data-wovnio-type=\"fallback_snippet\" async></script>"
+      else
+        "<script src=\"#{snippet_url}\" data-wovnio=\"#{data_wovnio}\" async></script>"
+      end
+    end
+
+    def data_wovnio
+      token = @store.settings['project_token']
+      current_lang = @headers.lang_code
+      default_lang = @store.settings['default_lang']
+      url_pattern = @store.settings['url_pattern']
+      lang_code_aliases_json = JSON.generate(@store.settings['custom_lang_aliases'])
+
+      CGI.escapeHTML(
+        [
+          "key=#{token}",
+          "backend=true",
+          "currentLang=#{current_lang}",
+          "defaultLang=#{default_lang}",
+          "urlPattern=#{url_pattern}",
+          "langCodeAliases=#{lang_code_aliases_json}",
+          "version=WOVN.rb"
```

Comment:
Can you also add the value at `VERSION` connstant, please?

Reply:
sure, I will

Reply:
thanks